### PR TITLE
Don't deselect my transaction when a new request comes in

### PR DESCRIPTION
**Pull request by philipkocanda**
## Summary

- Don't change $scope.activeKey when it's already set
- Update mock data every 5 seconds to simulate what's going on in many applications these days
- Bump version to 0.3.4

## Many words

When debugging an application where transactions keep coming in, RailsPanel would keep selecting the newest transaction. 

This is a bit annoying (understatement) when you're looking at a specific transaction to see what's going on. I fixed that by only automatically selecting a transaction when there is no transaction selected. I also added some code to the mock data function to simulate new transactions coming in every couple of seconds.

To test, open panel.html in Chrome and see the mock requests coming in, and the new requests not being selected (as they previously were).

### CODE REVIEW

File: rails_panel/assets/javascripts/panel.js [hunk 1]
```diff
@@ -7,19 +8,29 @@ var panel = {
...
+    this.addData(mockDataIndex + 3, scope, mockTransactions3());
+    this.addData(mockDataIndex + 4, scope, mockTransactions4());
+    mockDataIndex = mockDataIndex + 4;
+
+    // Simulate new requests coming in every couple of seconds
+    panel.mockDataTimer(scope);
+  },
+
+  mockDataTimer: function(scope) {
+    var timeout = setTimeout(
```
Comment (gogiel) [inline rails_panel/assets/javascripts/panel.js]: `timeout` is not used.

Comment (philipkocanda) [reply rails_panel/assets/javascripts/panel.js]: thanks :)

**Comment by gogiel** [comment]
LGTM. @dejan please review this as it also bumps the version.

**Comment by gogiel** [comment]
It would be nice if it was possible to deselect currently selected transaction.
Maybe it could be done by clicking the transaction again?

**Comment by philipkocanda** [comment]
Why would that be needed? You can simply select another transaction you are
interested in, or clear the view using CTRL/CMD+K.

**Comment by gogiel** [comment]
You're right, it won't work as I thought. 

What I mean is that the current behaviour might be still useful sometimes and I wonder if it was possible to still have that with some kind of a toggle.

**Comment by philipkocanda** [comment]
Might be nice to have indeed. Do you have an example of a case where the current behavior would be preferable?

**Comment by gogiel** [comment]
When working with the apps that don't pull data automatically you usually want to debug the latest request.

**Comment by gogiel** [comment]
I think in Rails Panel there should have a bar on top where all the toggles and switches could live.
It should look and behave like the bars in the other devtools tabs.
Screen Shot 2019-07-03 at 11 40 42

**Comment by philipkocanda** [comment]
That would be a cool feature :) Now that I know how this extension works, I might take some time to implement that! Though I think it's separate from this PR. @dejan what do you think?

**Comment by dejan** [comment]
This would break workflow for most cases. Usually it's the last requests that are of interest, but I understand how annoying must it be is if there's some polling happening. 

Without mechanism to switch between the two modes, I'm hesitant about the change.

**Pull request closed without merging**